misc/userscripts/qute-pass

```diff
@@ -173,6 +177,17 @@ def fake_key_raw(text):
         sequence = '" "' if character == ' ' else '\{}'.format(character)
         qute_command('fake-key {}'.format(sequence))
 
+def extract_password(secret, pattern):
+    match = re.match(pattern, secret)
+    if not match:
+        raise CouldNotMatchPassword
+    return match.group(1)
```

Comment:
You could take the opportunity here to fix a small bug that the current userscript has: if you provide a pattern that matches but with no capturing group, this will fail with `IndexError: no such group`. We could catch that exception and re-raise it as `CouldNotMatchPassword` as well.

Reply:
I compiled the regex pattern now, and check `prog.groups` which tells you the amount of capture groups the regex pattern has.
Example:
```
>>> re.compile("(.*)").groups                                                                 
1
>>> re.compile(".*").groups                                                                   
0
```
If `prog.groups` is atleast 1, the resulting match will never give IndexError when doing `match.group(1)`.
So we can now give user proper explanation that user have to supply a capture group in pattern:
```
    if not prog.groups:
        raise CouldNotMatchPassword("Pattern did not contain capture group, please use capture group. Example: (.*)")
```

Reply:
I thought the code might be more readable like this.
But, it might be better to just capture and rethrowing the `IndexError` instead of compiling the regex, and checking `compiled.group`
The only times `match.group(1)` can throw `IndexError` is when the regex lacks captureGroup.